docs/conceptual/core/storage.md

```diff
@@ -20,15 +20,21 @@ One advantage of using a simple key-value store is that you are able to easily a
 
 Substrate uses a Base-16 Modified Merkle Patricia tree ("trie") from [`paritytech/trie`](https://github.com/paritytech/trie) to provide a trie structure whose contents can be modified and whose root hash is recalculated efficiently.
 
-### Main Trie
+Tries are important tool for blockchains because the allows for efficient storing and sharing of historical of block state. You don't have a state trie per block, but a trie hash that will point to the nodes from previous block states. However, accessing data to trie data is costly. Each read operation takes O(log N) time, where N is the number of elements stored in the trie. To mitigate this, we use a key value cache.
 
-Substrate has a single main trie whose storage root hash is placed in the block header. This is used to easily verify the state of the blockchain and provide a basis for light clients to verify proofs.
+All trie node are stored in RocksDB and part of the trie state can get pruned, i.e. a key-value pair can be deleted from the storage when it is out of pruning range for non archive nodes. Trie nodes are also encoded to allow for any key type, while still be able to avoid key collision. We do not use [reference counting](http://en.wikipedia.org/wiki/Reference_counting) for performance reasons.
```

Comment:
Second sentence is a bit unclear, I would rephrase with:
"Trie nodes partial path is also prefixing the hash of encoded node in RocskDb keys, to avoid key collision."
May be a bit to technical (it assumes that readers know about traditional key construction (hash of the encoded node).